networkit/cpp/io/NetworkitBinaryReader.cpp

```diff
@@ -49,6 +49,21 @@ Graph NetworkitBinaryReader::readData(const T &source) {
         it += sizeof(uint64_t);
         memcpy(&header.features, it, sizeof(uint64_t));
         it += sizeof(uint64_t);
+        if(memcmp("nkbg003", header.magic, 8)) {
+            if(memcmp("nkbg002", header.magic, 8)) {
+                throw std::runtime_error("Reader expected another magic value");
+            } else {
+                directed = (header.features & nkbg::DIR_MASK);
+                weightFormat = static_cast<nkbg::WEIGHT_FORMAT>((header.features & nkbg::WGHT_MASK) >> nkbg::WGHT_SHIFT);
+                indexed = false;
+                version = 2;
+            }
+        } else {
+            directed = (header.features & nkbg::DIR_MASK);
+            weightFormat = static_cast<nkbg::WEIGHT_FORMAT>((header.features & nkbg::WGHT_MASK) >> nkbg::WGHT_SHIFT);
+            indexed = (header.features & nkbg::INDEX_MASK) >> nkbg::INDEX_SHIFT;
+            version = 3;
+        }
```

Comment:
There is some redundancy here. I think this part of the code should be restructured, for example like this:
```suggestion
if (!memcmp("nkbg002", header.magic, 8)) {
    version = 2;
}else if (!memcmp("nkbg003", header.magic, 8)) {
    version = 3;
}else{
    throw std::runtime_error("Reader expected another magic value");
}
directed = (header.features & nkbg::DIR_MASK);
weightFormat = static_cast<nkbg::WEIGHT_FORMAT>((header.features & nkbg::WGHT_MASK) >> nkbg::WGHT_SHIFT);
indexed = false;
if (version >= 3)
    indexed = (header.features & nkbg::INDEX_MASK) >> nkbg::INDEX_SHIFT;
```